### PR TITLE
Weather App project is finished.

### PR DESCRIPTION
**Pull request by RaminMammadzada**
The page is made with modular valinna JS (JS6) and the following milestones are finished:

 Milestone 1 - Setup HTML webpack to load style files, and pictures.
 Milestone 2 - Create an MVC structure to create view files, model to fetch data and controller to populate data to view via model.
 Milestone 3 - Setup Unsplash API and OpenWeather API to fetch data.
 Milestone 4 - Add extra spinning loader.

### CODE REVIEW

File: src/css/styles.css [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  align-items: center;
+}
+
+main {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+
+  /* background-color: #efeee8; */
```
Comment (christianotieno) [inline src/css/styles.css]: ### Remove commented out code
- [x] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: src/javascript/controller/weatherControler.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+  const setMainView = async () => {
+    locationNameView.innerHTML = '<div class="lds-ellipsis"><div></div><div></div><div></div><div></div></div>';
+    const weatherData = await fetchWeatherData(locationInput.value);
+    if (locationInput.value === '') {
+      locationNameView.innerText = 'You must enter a location name first!';
+    } else if (weatherData === null) {
+      locationNameView.innerText = 'The location couldnt found!';
+    } else {
+      // [, temperateView.innerText] = Math.round(temperatures);
```
Comment (christianotieno) [inline src/javascript/controller/weatherControler.js]: ### Remove commented out code
- [x] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: src/javascript/controller/weatherControler.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      [tempInC, tempInF, weatherCondition, countryAbbr] = weatherData;
+
+      temperateView.innerText = '';
+      weatherDescription.innerText = '';
+      weatherIcon.src = '';
+      await updateBodyBackground(weatherCondition.main);
+      await sleep(1500);
+      await setTemperatureView();
+      // console.log(">1> first run this");
```
Comment (christianotieno) [inline src/javascript/controller/weatherControler.js]: ### Remove commented out code
- [x] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: src/javascript/controller/weatherControler.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      event.preventDefault();
+      // Trigger the button element with a click
+      document.getElementById('search-button').click();
+    }
+  });
+
+  unitChanger.addEventListener('click', setTemperatureView);
+};
+
+// const populateFooter = () => {
```
Comment (christianotieno) [inline src/javascript/controller/weatherControler.js]: ### Remove commented out code
- [x] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: src/javascript/index.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import '@fortawesome/fontawesome-free/js/solid';
+import '@fortawesome/fontawesome-free/js/regular';
+import '@fortawesome/fontawesome-free/js/brands';
+import home from './view/home';
+import populateMain from './controller/weatherControler';
+
+home();
+
+populateMain();
+// fetchWeatherData();
```
Comment (christianotieno) [inline src/javascript/index.js]: ### Remove commented out code
- [x] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: src/javascript/model/imageData.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+const fetchImageData = async (queryText) => {
+  const result = await unsplash.search.getPhotos({
+    query: queryText,
+    page: 1,
+    perPage: 30,
+  });
+  const index = Math.floor(Math.random() * 30);
+  const img = result.response.results[index].links.download;
+  const avarageColorCode = result.response.results[index].color;
+  // console.log(">2> secondly run this");
```
Comment (christianotieno) [inline src/javascript/model/imageData.js]: ### Remove commented out code
- [x] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
+// Webpack uses this to work with directories
+const path = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+// const MiniCssExtractPlugin = require('mini-css-extract-plugin');
```
Comment (christianotieno) [inline webpack.config.js]: ### Remove commented out code
- [x] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: webpack.config.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+  output: {
+    path: path.resolve(__dirname, 'dist'),
+    filename: './bundle.js',
+  },
+  plugins: [
+    new HtmlWebpackPlugin({
+      hash: false,
+      filename: './index.html',
+    }),
+    // new MiniCssExtractPlugin()
```
Comment (christianotieno) [inline webpack.config.js]: ### Remove commented out code
- [x] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: webpack.config.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+      // {
+      //   test: /\.js$/,
+      //   exclude: /(node_modules)/,
+      //   use: {
+      //     loader: 'babel-loader',
+      //     options: {
+      //       presets: ['@babel/preset-env']
+      //     }
+      //   }
+      // },
```
Comment (christianotieno) [inline webpack.config.js]: ### Remove commented out code
- [x] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: webpack.config.js [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+      {
+        test: /\.css$/i,
+        use: [
+          // {
+          //   loader: MiniCssExtractPlugin.loader,
+          //   options: {
+          //     publicPath: ''
+          // },
+          //  },
```
Comment (christianotieno) [inline webpack.config.js]: ### Remove commented out code
- [x] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

**Review by christianotieno — CHANGES_REQUESTED** [review]
## Changes Required ♻️ 
Hello @RaminMammadzada 👋 
Congratulation on building this amazing weather app. It is neatly done clearly shows you have mastered the main concepts been tested for this project. the app is greatly modularised which is a plus and the code structure is neatly done. keep it up 👏 

- But before we proceed kindly take care of the following glaring issues I have highlighted for you down below 👇 

### Happy coding 💻 
#### Christian Otieno (CR)

File: src/css/styles.css (hunk 1 above)
Comment (RaminMammadzada) [reply src/css/styles.css]: Done.

File: src/javascript/controller/weatherControler.js (hunk 2 above)
Comment (RaminMammadzada) [reply src/javascript/controller/weatherControler.js]: Done.

File: src/javascript/controller/weatherControler.js (hunk 3 above)
Comment (RaminMammadzada) [reply src/javascript/controller/weatherControler.js]: Done.

File: src/javascript/controller/weatherControler.js (hunk 4 above)
Comment (RaminMammadzada) [reply src/javascript/controller/weatherControler.js]: Done.

File: src/javascript/index.js (hunk 5 above)
Comment (RaminMammadzada) [reply src/javascript/index.js]: Done.

File: src/javascript/model/imageData.js (hunk 6 above)
Comment (RaminMammadzada) [reply src/javascript/model/imageData.js]: Done.

File: webpack.config.js (hunk 7 above)
Comment (RaminMammadzada) [reply webpack.config.js]: Done.

File: webpack.config.js (hunk 8 above)
Comment (RaminMammadzada) [reply webpack.config.js]: Done.

File: webpack.config.js (hunk 9 above)
Comment (RaminMammadzada) [reply webpack.config.js]: Done.

File: webpack.config.js (hunk 10 above)
Comment (RaminMammadzada) [reply webpack.config.js]: Done.

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @RaminMammadzada 
### Good work you have done so far :clap: You have a minor change to make
- [x] We recommend the use of es6 arrow functions rather than the function keyword, example `const invertColor = (hex, be) => { # Your code }`
- [ Optional ] You can speed up the loading time before the data is displayed.
### Make the changes and submit for another code review. Good luck.
- Happy coding!

**Review by christianotieno — APPROVED** [review]
## Project Approved 🍾 
Hello @RaminMammadzada 👋 
Good job working on the changes above 👆 
The project's been approved. All the best in your next tasks 🥇 
## Happy coding 💻 
#### Christian Otieno (CR)

**Pull request merged**